evmrpc/filter.go

```diff
@@ -676,6 +680,31 @@ func (f *LogFetcher) GetLogsByFilters(ctx context.Context, crit filters.FilterCr
 	if lastToHeight > begin {
 		begin = lastToHeight
 	}
+	if begin < earliest {
+		begin = earliest
```

Comment:
I'd propose that we should fail with an error like the block range is too far out of range rather than quietly succeeding with a different range.